wire-schema/src/jvmMain/java/com/squareup/wire/schema/PruningRules.kt

```diff
@@ -110,104 +109,105 @@ class PruningRules private constructor(builder: Builder) {
-      usedIncludes.add(includeMatch!!)
-    } else if (excludeMatch != null) {
-      usedExcludes.add(excludeMatch)
+      usedRoots.add(rootMatch!!)
+    } else if (pruneMatch != null) {
+      usedPrunes.add(pruneMatch)
     }
     return isRoot
   }
 
   /**
-   * Returns true if `type` should be excluded, even if it is a transitive dependency of a root. In
-   * that case, the referring member is also excluded.
+   * Returns true if `type` should be pruned, even if it is a transitive dependency of a root. In
+   * that case, the referring member is also pruned.
    */
-  fun excludes(type: ProtoType) = exclude(type.toString())
+  fun prunes(type: ProtoType) = prunes(type.toString())
 
-  /** Returns true if `protoMember` should be excluded. */
-  fun excludes(protoMember: ProtoMember) = exclude(protoMember.toString())
+  /** Returns true if `protoMember` should be pruned. */
+  fun prunes(protoMember: ProtoMember) = prunes(protoMember.toString())
 
-  /** Returns true if `identifier` or any of its enclosing identifiers is excluded.  */
-  private fun exclude(identifier: String): Boolean {
-    var includeMatch: String? = null
-    var excludeMatch: String? = null
+  /** Returns true if `identifier` or any of its enclosing identifiers is pruned.  */
+  private fun prunes(identifier: String): Boolean {
+    var rootMatch: String? = null
+    var pruneMatch: String? = null
     var rule: String? = identifier
     while (rule != null) {
-      if (excludeMatch == null && excludes.contains(rule)) {
-        excludeMatch = rule
+      if (pruneMatch == null && prunes.contains(rule)) {
+        pruneMatch = rule
       }
-      if (includeMatch == null && includes.contains(rule)) {
-        includeMatch = rule
+      if (rootMatch == null && roots.contains(rule)) {
+        rootMatch = rule
       }
       rule = enclosing(rule)
     }
 
-    val excluded = when {
-      excludeMatch != null && includeMatch != null -> excludeMatch.length >= includeMatch.length
-      excludeMatch != null -> true
+    val pruned = when {
+      pruneMatch != null && rootMatch != null -> pruneMatch.length >= rootMatch.length
+      pruneMatch != null -> true
       else -> false
     }
-    if (excluded) {
-      usedExcludes.add(excludeMatch!!)
+    if (pruned) {
+      usedPrunes.add(pruneMatch!!)
     }
-    return excluded
+    return pruned
   }
 
-  fun unusedIncludes() = includes - usedIncludes
+  fun unusedRoots() = roots - usedRoots
 
-  fun unusedExcludes() = excludes - usedExcludes
+  fun unusedPrunes() = prunes - usedPrunes
 
   class Builder {
-    internal val includes = mutableSetOf<String>()
-    internal val excludes = mutableSetOf<String>()
+    internal val roots = mutableSetOf<String>()
+    internal val prunes = mutableSetOf<String>()
     internal var oldest: SemVer? = null
     internal var newest: SemVer? = null
 
-    fun include(identifier: String) = apply {
-      includes.add(identifier)
+    fun root(identifier: String) = apply {
```

Comment:
While I think that `prune()` is a good method name, I'd prefer this to be called `addRoot()`. The meaning of the verb "root" is not very clear, besides I feel like this can be confused as "mark this identifier as _the_ root, unmarking the previously marked one".

Reply:
fixed